Expose flag for critical extensions (#4568)
ARMmbed/mbedtls: ChangeLog.d/issue4055.txt

API changes

>    * Add parameter to mbedtls_x509write_csr_set_extension which allows to mark
>      an extension as critical. Fixes #4055.

Reviewer: ```suggestion
   * The function mbedtls_x509write_csr_set_extension() has an extra parameter
     which allows marking an extension as critical. Fixes #4055.
```
